src/common/graphHelpers.js

```diff
@@ -1,45 +1,97 @@
 import moment from 'moment';
 import { rhelApiTypes } from '../types/rhelApiTypes';
-import { helpers } from './helpers';
+import { translate } from '../components/i18n/i18n';
+
+const GRANULARITY_TYPES = rhelApiTypes.RHSM_API_QUERY_GRANULARITY_TYPES;
 
 /**
  * Chart Date Format (used in axis and tooltips)
  */
-const chartDateFormat = 'MMM D';
+const chartDateDayFormatShort = 'MMM D';
+const chartDateDayFormat = 'MMM D YYYY';
+
+const chartDateMonthFormatShort = 'MMM';
+const chartDateMonthFormat = 'MMM YYYY';
 
 /**
- * Generate a fallback graph with zeroed data
+ * Returns x axis ticks/intervals array for the xAxisTickInterval
  *
- * @param startDate {string}
- * @param endDate {string}
- * @returns {Array}
+ * @param {string} granularity, see enum of rhelApiTypes.RHSM_API_QUERY_GRANULARITY_TYPES
+ * @returns {number}
  */
-const zeroedUsageData = (startDate, endDate) => {
-  const zeroedArray = [];
-  const endDateStartDateDiff = moment(endDate).diff(startDate, 'days');
+const getChartXAxisLabelIncrement = granularity => {
+  switch (granularity) {
+    case GRANULARITY_TYPES.DAILY:
+      return 5;
+    case GRANULARITY_TYPES.WEEKLY:
+      return 2;
+    case GRANULARITY_TYPES.MONTHLY:
+    case GRANULARITY_TYPES.QUARTERLY:
+    default:
+      return 1;
+  }
+};
 
-  for (let i = 0; i <= endDateStartDateDiff; i++) {
-    const clonedStartDate = moment.utc(startDate);
-    zeroedArray.push({
-      x: clonedStartDate.add(i, 'days').format(chartDateFormat),
-      y: 0
-    });
+/**
+ * Return translated labels based on granularity.
+ *
+ * @param {string} granularity, see enum of rhelApiTypes.RHSM_API_QUERY_GRANULARITY_TYPES
+ * @returns {Object}
+ */
+const getGraphLabels = granularity => {
+  const labels = {
+    label: translate('curiosity-graph.tooltipLabel')
+  };
+
+  switch (granularity) {
+    case GRANULARITY_TYPES.WEEKLY:
+      labels.previousLabel = translate('curiosity-graph.tooltipPreviousLabelWeekly');
+      break;
+    case GRANULARITY_TYPES.MONTHLY:
+      labels.previousLabel = translate('curiosity-graph.tooltipPreviousLabelMonthly');
+      break;
+    case GRANULARITY_TYPES.QUARTERLY:
+      labels.previousLabel = translate('curiosity-graph.tooltipPreviousLabelQuarterly');
+      break;
+    case GRANULARITY_TYPES.DAILY:
+    default:
+      labels.previousLabel = translate('curiosity-graph.tooltipPreviousLabelDaily');
+      break;
   }
 
-  return zeroedArray;
+  return labels;
+};
+
+/**
+ * Return a time allotment based on granularity
+ *
+ * @param {string} granularity enum based on rhelApiTypes.RHSM_API_QUERY_GRANULARITY_TYPES
+ * @returns {string}
+ */
+const getGranularityDateType = granularity => {
+  switch (granularity) {
+    case GRANULARITY_TYPES.DAILY:
+      return 'days';
```

Comment:
are these literals backend API constants? Maybe move this helper function into `rhelApiTypes` file instead of graphHelpers? Not really sure, just seems different.

Reply:
so the rhelApiTypes are meant to be static... this is geared towards a specific date type from moment

Reply:
if this type switch is used beyond the immediate helper, then willing to reconsidering moving it within a type file, or within the `dateHelpers.js`